source/common/tls/utility.h

```diff
@@ -88,6 +88,13 @@ std::string getIssuerFromCertificate(X509& cert);
  */
 std::string getSubjectFromCertificate(X509& cert);
 
+/**
+ * Retrieves the extension OIDs from certificate.
+ * @param cert the certificate
+ * @return std::vector returns the string list of ASN.1 object identifiers.
+ */
+std::vector<std::string> getCertificateExtensionOids(X509& cert);
```

Comment:
Can you give an example of OID value here? I know that GKE puts pod IDs as OIDs but I can't quite map the value representation to this API.

Reply:
This API returns the list of OID strings, not the extension value OID associated with. As shown in the test case:

https://github.com/envoyproxy/envoy/pull/35953/files#diff-45b4e28019cff37f48b67bdddfd885a539902471b71c6720b91b6a027acc10e4R189-R198

It is intended to enable Lua API user to detect whether a specific extension exist from the peer cert, and use that information for further business logic.

Reply:
Is there an ABI to get the value of OID?